packages/bun-vscode/src/features/tests/bun-test-controller.ts

```diff
@@ -0,0 +1,1325 @@
+    content: string,
+    index: number,
+    type: TestNode["type"],
+    line: number,
+  ): TestNode[] {
+    if (!fullMatch.includes(".each")) {
+      return [
+        {
+          name: name.replace(/\\/g, ""),
+          type,
+          line,
+          children: [],
+          startIdx: index,
+        },
+      ];
+    }
+
+    const eachMatch = content.slice(index).match(/\.each\s*\(\s*(\[[\s\S]*?\])\s*\)/);
+    if (!eachMatch) {
+      return [
+        {
+          name: name.replace(/\\/g, ""),
+          type,
+          line,
+          children: [],
+          startIdx: index,
+        },
+      ];
+    }
+
+    const arrayString = eachMatch[1].replace(/,\s*(?=[\]\}])/g, "");
+
+    try {
+      const eachValues = JSON.parse(arrayString);
+      if (!Array.isArray(eachValues)) {
+        throw new Error("Not an array");
+      }
+
+      return eachValues.map(val => {
+        let testName = name;
+        if (Array.isArray(val)) {
+          let idx = 0;
+          testName = testName.replace(/%[isfd]/g, () => {
+            const v = val[idx++];
+            return typeof v === "object" ? JSON.stringify(v) : String(v);
+          });
+        } else {
+          testName = testName.replace(/%[isfd]/g, () => {
+            return typeof val === "object" ? JSON.stringify(val) : String(val);
+          });
+        }
+
+        return {
+          name: testName,
+          type,
+          line,
+          children: [],
+          startIdx: index,
+        };
+      });
+    } catch {
+      return [
+        {
+          name: name.replace(/\\/g, ""),
+          type,
+          line,
+          children: [],
+          startIdx: index,
+        },
+      ];
+    }
+  }
+
+  private addTestNodes(nodes: TestNode[], parent: vscode.TestItem, filePath: string, parentPath = ""): void {
+    for (const node of nodes) {
+      const nodePath = parentPath
+        ? `${parentPath} > ${this.escapeTestName(node.name)}`
+        : this.escapeTestName(node.name);
+      const testId = `${filePath}#${nodePath}`;
+
+      const testItem = this.testController.createTestItem(testId, this.stripAnsi(node.name), vscode.Uri.file(filePath));
+
+      testItem.tags = [new vscode.TestTag(node.type === "describe" ? "describe" : "test")];
+
+      if (typeof node.line === "number") {
+        testItem.range = new vscode.Range(
+          new vscode.Position(node.line, 0),
+          new vscode.Position(node.line, node.name.length),
+        );
+      }
+
+      parent.children.add(testItem);
+
+      if (node.children.length > 0) {
+        this.addTestNodes(node.children, testItem, filePath, nodePath);
+      }
+      testItem.canResolveChildren = false;
+    }
+  }
+
+  private stripAnsi(source: string): string {
+    return source.replace(/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-PRZcf-nqry=><]/g, "");
+  }
+
+  private escapeTestName(source: string): string {
+    return source.replace(/[^a-zA-Z0-9_\ ]/g, "\\$&");
```

Comment:
Tests may have json in the name. This should just be a strip ansi IMO

Reply:
This escape is mainly for --test-name-patterm to remove special characters, and to keep ` > ` as the separator for the name and parent describes. I proabibly could have done a block list instead though.